lang/validate.go

```diff
@@ -9,4 +9,21 @@ import (
 	"github.com/hashicorp/hcl/v2"
 )
 
-type ValidationFunc func(ctx context.Context) hcl.Diagnostics
+type ValidationFunc func(ctx context.Context) DiagnosticsMap
+
+type DiagnosticsMap map[string]hcl.Diagnostics
+
+func (dm DiagnosticsMap) Extend(diagMap DiagnosticsMap) DiagnosticsMap {
+	for fileName, diags := range diagMap {
+		_, ok := dm[fileName]
+		if !ok {
+			dm[fileName] = make(hcl.Diagnostics, 0)
+		}
+
+		// TODO: this seems to wipe instead of extend
+		// dm[fileName].Extend(diags)
+		dm[fileName] = append(dm[fileName], diags...)
```

Comment:
@radeksimko can you take a look at whether this should stay as you intended or should this change stay?